velox/dwio/common/SelectiveStructColumnReader.cpp

```diff
@@ -129,8 +129,7 @@ void SelectiveStructColumnReaderBase::next(
     resultRowVector->unsafeResize(numValues);
 
     for (auto& childSpec : scanSpec_->children()) {
-      VELOX_CHECK(childSpec->isConstant());
```

Comment:
We should not go through if there is explicit row number column, no?

Reply:
This is captured by the test case: https://github.com/facebookincubator/velox/blob/17c3d8480ffee051252dcb7b61ce1fafcb0ec6ba/velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp#L624.
I think both constant columns and explicit row number column would not generate `SelectiveColumnReader`, so when `children_.empty()`, we need to also handle explicit row number column here. Please let me know if anything I miss here.

Reply:
Then let's just do `VELOX_CHECK(childSpec->isConstant() || childSpec->isExplicitRowNumber())`

Reply:
Sure, updated.